javagen/src/main/java/com/azure/autorest/util/MethodUtil.java

```diff
@@ -55,4 +70,150 @@ public static boolean isMethodIncludeRepeatableRequestHeaders(ProxyMethod proxyM
     public static boolean isHttpMethodSupportRepeatableRequestHeaders(HttpMethod httpMethod) {
         return REPEATABILITY_REQUEST_HTTP_METHODS.contains(httpMethod);
     }
+
+    /**
+     * @param requests a list of requests
+     * @return the first request consumes binary type, if no binary request, return the first request in requests
+     *
+     * If the binary request does not have content-type parameter, we will add one for it
+     */
+    public static Request findFirstBinaryRequest(List<Request> requests, Operation operation) {
+        Request selectedRequest = requests.get(0);
+        for (Request request : requests) {
+            if (request.getProtocol().getHttp().getKnownMediaType() != null
+                    && request.getProtocol().getHttp().getKnownMediaType().equals(KnownMediaType.BINARY)) {
+                // add contentType parameter
+                if (haveDifferentContentTypes(requests) && !hasContentTypeParameter(request)) {
+                    Parameter contentTypeParameter = createContentTypeParameter(request, operation);
+                    request.getParameters().add(getBinarySchemaBodyOrContentLengthParameterIndex(request.getParameters()), contentTypeParameter);
+                    if (contentTypeParameter.isRequired()) {
+                        request.getSignatureParameters().add(getBinarySchemaBodyOrContentLengthParameterIndex(request.getSignatureParameters()), contentTypeParameter);
+                    }
+                }
+                selectedRequest = request;
+                break;
+            }
+        }
+        return selectedRequest;
+    }
+
+    /**
+     *
+     * @param request the input request
+     * @return true if there is parameter in the request named "contentType", otherwise, return false
+     */
+    private static boolean hasContentTypeParameter(Request request) {
+        for (Parameter parameter : request.getParameters()) {
+            if (parameter.getLanguage().getJava().getSerializedName() != null
```

Comment:
I don't think you need to protect `parameter.getLanguage().getJava().getSerializedName() != null`, but you might want protect `parameter.getLanguage().getJava() != null` insteadl.